### PR TITLE
HARMONY-1938: Support running harmony without using an EDL app

### PR DESCRIPTION
**Pull request by chris-durbin**
## Jira Issue ID
HARMONY-1938

## Description
This change allows developers to run harmony without having to set up an EDL application.

## Local Test Steps
There are several configurations to test.
1. Development with USE_EDL_CLIENT_APP=true and OAUTH* variables configured.
2. Development with USE_EDL_CLIENT_APP=false and OAUTH* variables NOT configured.
3. Harmony in a box with USE_EDL_CLIENT_APP=true and OAUTH* variables configured.
4. Harmony in a box with USE_EDL_CLIENT_APP=false and OAUTH* variables NOT configured.

Note any services using the current harmony service library will fail if OAUTH* environment variables are not set. In order to test without an EDL client app you will need to build the service using the harmony-service-library from the harmony-1938 branch.

### Test bin/bootstrap and/or bin/create-dotenv with no .env file.
* Verify that you are prompted to choose whether or not to use a client app and the default is to not use the app.
* Verify that if you choose to not use an app you are prompted for an EDL token.
* Verify if you choose to use an app you are prompted for OAUTH* parameters.

### Other Tests
* Verify if USE_EDL_CLIENT_APP=false and EDL_TOKEN is not provided harmony throws an error when starting up
* Verify if USE_EDL_CLIENT_APP=true and any OAUTH* environment variables are not set harmony throws an error when starting up
* Verify if the EDL_TOKEN is invalid the endpoints for submitting requests returns a 40x error

### Endpoints that should return authorization errors when not using an EDL_CLIENT_APP
* /service-image-tag
* /service-deployment
* /service-deployments-state
* /admin/jobs
* /admin/workflow-ui
* /admin/health
* /admin/request-metrics
* /core/configuration/log-level

Also verify that the workflow-ui does not show the logs button.

Verify that all other endpoints and functionality work as expected.

## PR Acceptance Checklist
* [X] Acceptance criteria met
* [X] Tests added/updated (if needed) and passing
* [X] Documentation updated (if needed)
* [X] Harmony in a Box tested? (if changes made to microservices)

### CODE REVIEW

File: services/harmony/app/middleware/cmr-collection-reader.ts [hunk 1]
```diff
@@ -39,26 +40,28 @@ async function loadVariablesForCollection(context: RequestContext, collection: C
...
-        const eulaInfo: EdlUserEulaInfo = await verifyUserEula(req.context, req.user, eulaId);
-        if (eulaInfo.statusCode == 404 && eulaInfo.acceptEulaUrl) { // EULA wasn't accepted
-          acceptEulaUrls.push(eulaInfo.acceptEulaUrl);
-        } else if (eulaInfo.statusCode == 404) {
-          req.context.logger.error(`EULA (${eulaId}) verfification failed with statusCode 404. Error: ${eulaInfo.error}`);
-          throw new NotFoundError(`EULA ${eulaId} could not be found.`);
-        } else if (eulaInfo.statusCode !== 200) {
-          req.context.logger.error(`EULA (${eulaId}) verfification failed. Error: ${eulaInfo.error}`);
-          throw new ServerError(`EULA (${eulaId}) verfification failed unexpectedly.`);
+  if (env.useEdlClientApp) {
```
Comment (indiejames) [inline services/harmony/app/middleware/cmr-collection-reader.ts:43]: I don't understand what whether or not you are a EDL client app has to do with whether or not a user has accepted eulas.

Comment (chris-durbin) [reply services/harmony/app/middleware/cmr-collection-reader.ts:43]: It requires making a call to EDL that requires a client app so we can't do the check.

**Comment by ygliuvt** [comment]
nit: Maybe consider adding a check in `bin/bootstrap-harmony` to halt the bootstrap with a reasonable error message when there is no EDL_TOKEN configured and USE_EDL_CLIENT_APP is false rather than the current behavior where the services are started up and harmony is not working. I have to go to harmony pod log to see the following error:
```
Error: Earthdata Login configuration error: You must set EDL_TOKEN in the environment when USE_EDL_CLIENT_APP is false
```

**Review by ygliuvt — APPROVED** [review]
Tested successfully with the harmony-service-example service image built with the harmony-1938 branch of harmony-service-lib-py. 
Tested with with and without USE_EDL_CLIENT_APP configuration, harmony dev and harmony-in-a-box locally. Verified service example requests work in all conditions and the admin routes are disabled with permission error when USE_EDL_CLIENT_APP is set to false.

**Review by indiejames — APPROVED** [review]
Tested locally using both local development services and Harmony in a box

**Pull request merged**